theano/tensor/basic.py

```diff
@@ -5943,25 +5981,77 @@ def infer_shape(self, node, shapes):
 
 
 def diagonal(a, offset=0, axis1=0, axis2=1):
```

Comment:
We talked about it in private. For the record, ```offset=0``` could be replace by something like ```offset=Diagonal.default_offset``` and similarly for axis1 and axis2.